source/fundamentals/typescript.txt

```diff
@@ -79,81 +84,119 @@ You can find a code snippet that shows how to specify a type for the ``FindCurso
-access nested fields in query and update documents when you use TypeScript. You
-must manually check that your nested field values have your intended type.
+The following code snippet references a top-level instance of the
+``RecursivePet`` interface with an incorrect type and raises a type error:
 
-.. note:: Reason To Use Dot Notation
+.. tabs::
 
-   In the MongoDB Query Language, you must match a subdocument exactly
-   when specifying subdocuments in a query. Dot notation allows you to query
-   nested fields without matching subdocuments exactly.
+   .. tab:: Code Snippet
+      :tabid: code-snippet
 
-   To show this behavior, lets say you have a collection containing
-   only the following document:
+      .. code-block:: typescript
+         :emphasize-lines: 3
 
-   .. code-block:: json
+         database
+           .collection<RecursivePet>("your-collection")
+           .findOne({ pet: "Spot" });
 
-      { field: { s1: "hi", s2: "bye" } }
+   .. tab:: Error
+      :tabid: error
 
-   The following query returns no results from this collection, as the value of
-   ``field`` does not exactly match ``{ s1: "hi" }``:
+      .. code-block:: text
+                  
+         index.ts(19,59): error TS2769: No overload matches this call.
+         The last overload gave the following error.
+         Type 'string' is not assignable to type 'Condition<Pet>'.
 
-   .. literalinclude:: /code-snippets/typescript/note-on-dot-notation.ts
-      :language: typescript
-      :linenos:
-      :start-after: start-no-doc
-      :end-before: end-no-doc
+If you must have type safety within nested instances of recursive types,
+you must write your query or update without dot notation.
 
-   The following queries both return your document:
+To learn more about dot notation, see 
+:manual:`Dot Notation </core/document/#dot-notation>`
+in the MongoDB manual.
 
-   .. literalinclude:: /code-snippets/typescript/note-on-dot-notation.ts
-      :language: typescript
-      :linenos:
-      :start-after: start-doc
-      :end-before: end-doc
+.. _node-driver-limitations-mutual-recursion:
 
-   The syntax of the query that does not use dot notation is cumbersome and hard
-   to understand, and may not be worth the type safety obtained from
-   avoiding dot notation.
+Mutual Recursion
+~~~~~~~~~~~~~~~~
+
+You cannot specify a mutually recursive type as a type parameter in version
+{+version+} of the driver.
+
+If you specify a mutually recursive type, the TypeScript compiler raises the
+following error:
+
+.. code-block:: text
+
+   error TS2615: Type of property 'r' circularly references itself in mapped type '{ [Key in keyof MutuallyRecursive]: MutuallyRecursive[Key] extends MutuallyRecursive ? [Key] : MutuallyRecursive extends MutuallyRecursive[Key] ? [...] : MutuallyRecursive[Key] extends readonly (infer ArrayType)[] ? MutuallyRecursive extends ArrayType ? [...] : ArrayType extends MutuallyRecursive ? [...] : [...] : [...'.
```

Comment:
Could you truncate the full error without losing any important information? For example:

> error TS2615: Type of property 'r' circularly references itself in mapped type...

Or is there something in the error message that you want the reader to notice? If so, could you include an explanation?

Reply:
Truncated the error message as suggested.